docs/developers/components/README.md

```diff
@@ -1,34 +1,268 @@
+Other features are provided by _custom components_. These can be npm packages such as [@harperdb/nextjs](https://github.com/HarperDB/nextjs) and [@harperdb/apollo](https://github.com/HarperDB/apollo) (which are maintained by Harper), or something maintained by the community. Custom components follow the same configuration rules and use the same APIs that Harper's internal components do. The only difference is that they must be apart of the component's dependencies.
 
-Server resource components implement support for different types of files that can be used as resources in applications. HarperDB includes support for using JavaScript modules and GraphQL Schemas as resources, but resource components may add support for different file types like HTML templates (like JSX), CSV data, and more.
+> Documentation is available for all [internal](#internal-components) and [custom](#custom-components) Harper components.
 
-## Server components
+<!-- TODO: add a callout to a list of third-party components here. Maybe also a link to something like an awesome-harper for community things? -->
 
-Server components can be easily be added and configured by simply adding an entry to your harperdb-config.yaml:
+## Component Configuration
+
+Harper components are configured with a **config.yaml** file located in the root of the component module directory. This file is how a component configures other components it depends on. Each entry in the file starts with a component name, and then configuration values are indented below it.
+
+```yaml
+name:
+  option-1: value
+  option-2: value
+```
+
+It is the entry's **name** that is used for component resolution. It can be one of the [internal components](#internal-components), or it must match a package dependency of the component as specified by **package.json**. The [Custom Component Configuration](#custom-component-configuration) section provides more details and examples.
+
+For some [internal components](#internal-components), they can sometimes be configured with as little as a top-level boolean; for example, the [rest](#rest) extension can be enabled with just:
+
+```yaml
+rest: true
+```
+
+Other components (internal or custom), will generally have more configuration options. Some options are ubiquitous to the Harper platform, such as the `files`, `path`, and `root` options for a [Resource Extension](#resource-extension-configuration), or `package` for a [custom component](#custom-component-configuration). Additionally, [custom options](#protocol-extension-configuration) can be defined for [Protocol Extensions](#protocol-extension).
+
+### Custom Component Configuration
+
+Any custom component **must** be configured with the `package` option in order for Harper to load that component. When enabled, the name of package must match a dependency of the component. For example, to use the `@harperdb/nextjs` extension, it must first be included in **package.json**:
```

Comment:
@Ethan-Arrowood want to link to the repo here for convenience if someone browsing the docs wants to learn more about this extension specifically? https://github.com/HarperDB/nextjs